src/components/widgets/TableWidget/TableWidget.tsx

```diff
@@ -55,7 +55,7 @@ export interface TableWidgetProps extends TableWidgetOwnProps {
     filterGroups: FilterGroup[],
     onDrillDown: (widgetName: string, bcName: string, cursor: string, fieldKey: string) => void,
     onShowAll: (bcName: string, cursor: string, route: Route, widgetName: string) => void,
-    onOperationClick: (bcName: string, operationType: string, widgetName: string) => void,
+    onOperationClick: (bcName: string, operationType: string, widgetName: string, bcKey?: string, preInvoke?: OperationPreInvoke) => void,
```

Comment:
We should avoid complicating functions contract when possible.  
There are already a `widgetName` and `operationType` arguments, you can use them inside the middleware to find the widget, the operation and get `bcKey` and `preInvoke` from there

Reply:
fixed. Use from state.view.rowMeta